File: Champions/Global/Recall.lua
Vector2 = require 'Vector2' -- include 2d vector lib 

function onStartCasting()      
end

function onFinishCasting()
	--addBuff("Recall", 8, owner, owner)
	addParticleTarget(owner, "TeleportHome.troy", owner)
end

function onStartChanneling()
  printChat("StartChanneling")
end

function onFinishChanneling()             
  printChat("FinishChanneling")
  local teamBlue = Vector2:new(26, 280)       
  local teamPurple = Vector2:new(13927, 14175)      
  teleport(owner, teamBlue.x, teamBlue.y)
end

function applyEffects()
end

function onUpdate(diff)
end

File: Champions/Global/Recall.cs
using GameServerCore.Domain.GameObjects;
using GameServerCore.Domain;
using LeagueSandbox.GameServer.Scripting.CSharp;

namespace Spells
{
    public class Recall : IGameScript
    {
        public void OnStartCasting(IChampion owner, ISpell spell, IAttackableUnit target)
        {
        }

        public void OnFinishCasting(IChampion owner, ISpell spell, IAttackableUnit target)
        {
            // @TODO Find out if possible to check for champion movement
            owner.AddBuffGameScript("Recall", "Recall", spell, 8.0f, true);
        }

        public void ApplyEffects(IChampion owner, IAttackableUnit target, ISpell spell, IProjectile projectile)
        {
        }

        public void OnUpdate(double diff)
        {
        }

        public void OnActivate(IChampion owner)
        {

        }

        public void OnDeactivate(IChampion owner)
        {

        }
    }
}

